index.js

```diff
@@ -66,65 +67,95 @@ var Converter = function() {};
  * @param apiDeclarations {array} - a list of resources
  * @returns {object} - Fully converted Swagger 2.0 document
 */
-Converter.prototype.convert = function(resourceListing, apiDeclarations) {
+prototype.convert = function(resourceListing, apiDeclarations) {
   assert(typeof resourceListing === 'object');
 
   var convertedSecurityNames = {};
   var securityDefinitions = this.buildSecurityDefinitions(resourceListing,
     convertedSecurityNames);
 
-  var tagDescriptions = {};
-  this.forEach(resourceListing.apis, function(resource) {
-    var tagName = this.extractTag(resource.path);
-    if (!isValue(tagName)) { return; }
-
-    tagDescriptions[tagName] = resource.description;
-  });
-
   var tags = [];
-  var paths = {};
-  var definitions = {};
-  var pathComponents = {};
+  this.paths = {};
+  this.definitions = {};
 
-  // Handle embedded documents
-  var resources = [resourceListing];
-  if (isValue(apiDeclarations)) {
-    resources = resources.concat(apiDeclarations);
+  if (this.isEmbeddedDocument(resourceListing)) {
+    this.convertApiDeclaration(resourceListing, undefined);
   }
+  else {
+    tags = this.buildTags(resourceListing, apiDeclarations);
 
-  this.forEach(resources, function(resource) {
-    var operationTags;
-    var tagName = this.extractTag(resource.resourcePath);
+    this.forEach(apiDeclarations, function(declaration, index) {
+      var operationTags;
 
-    if (isValue(tagName)) {
-      tags.push(extend({}, {
-        name: tagName,
-        description: tagDescriptions[tagName]
-      }));
-      operationTags = [tagName];
+      var tag = tags[index];
+      if (isValue(tag)) {
+        operationTags = [tag.name];
+      }
+
+      this.convertApiDeclaration(declaration, operationTags);
+    });
+  }
+
+  return extend({},
+    this.aggregatePathComponents(resourceListing, apiDeclarations),
+    {
+      swagger: '2.0',
```

Comment:
Are we removing extra keys here if original swagger does not have those features?